File: app/src/main/java/com/ad/whatsappclone/activity/WelcomeActivity.java
package com.ad.whatsappclone.activity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.ad.whatsappclone.databinding.ActivityWelcomeBinding;

public class WelcomeActivity extends AppCompatActivity {
    ActivityWelcomeBinding binding;

    @Override
    protected void onStart() {
        super.onStart();
        if (getIntroPrefData()) {
            startActivity(new Intent(WelcomeActivity.this, SignInActivity.class));
            finish();
        }
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityWelcomeBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        if (getSupportActionBar() != null) {
            getSupportActionBar().hide();
        }
    }

    public void onAgreeBtnClick(View view) {
        Intent intent = new Intent(WelcomeActivity.this, SignInActivity.class);
        startActivity(intent);
        finish();
        saveIntroData();
    }

    // TODO:shared preference data
    private void saveIntroData() {
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences("Welcome", MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean("isWelcomeDone", true);
        editor.apply();
    }

    private boolean getIntroPrefData() {
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences("Welcome", MODE_PRIVATE);
        return sharedPreferences.getBoolean("isWelcomeDone", false);
    }
}

File: app/src/main/java/com/ad/whatsappclone/ui/authentication/WelcomeActivity.kt
package com.ad.whatsappclone.ui.authentication

import android.content.Intent
import android.os.Bundle
import com.ad.whatsappclone.databinding.ActivityWelcomeBinding
import com.ad.whatsappclone.ui.BaseActivity

class WelcomeActivity : BaseActivity() {
    private lateinit var binding: ActivityWelcomeBinding

    override fun onStart() {
        super.onStart()
        if (introPrefData) {
            startActivity(Intent(this@WelcomeActivity, SignInActivity::class.java))
            finish()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityWelcomeBinding.inflate(layoutInflater)
        setContentView(binding.root)
        hideActionBar()
        binding.btnAgreeContinue.setOnClickListener {
            onAgreeBtnClick()
        }
    }

    private fun onAgreeBtnClick() {
        val intent = Intent(this@WelcomeActivity, PhoneAuthActivity::class.java)
        startActivity(intent)
        finish()
        saveIntroData()
    }

    // TODO:shared preference data
    private fun saveIntroData() {
        val sharedPreferences = applicationContext.getSharedPreferences("Welcome", MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putBoolean("isWelcomeDone", true)
        editor.apply()
    }

    private val introPrefData: Boolean
        get() {
            val sharedPreferences = applicationContext.getSharedPreferences("Welcome", MODE_PRIVATE)
            return sharedPreferences.getBoolean("isWelcomeDone", false)
        }
}